rrdd/design/plugin-protocol-v3.md

```diff
@@ -0,0 +1,81 @@
+---
+title: RRDD plugin protocol v3
+layout: default
+design_doc: true
+revision: 1
+status: proposed
+revision_history:
+- revision_number: 1
+  description: Initial version
+---
+
+Motivation
+----------
+
+rrdd plugins protocol v2 report datasources via shared-memory file, however it
+has various limitations :
+ - metrics are unique by their names, thus it is not possible cannot have
+  several metrics that shares a same name (e.g vCPU usage per vm)
+ - only number metrics are supported, for example we can't expose string
```

Comment:
What's the utility of having cpu models exposed in the metrics interface? this isn't supposed to change